stacks/analytics-ingest-lambda.yml

```diff
@@ -0,0 +1,142 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: Analytics ingest lambda functions
+Parameters:
+  OpsErrorMessagesSnsTopicArn:
+    Type: String
+  CodeS3Bucket:
+    Type: String
+  CodeS3ObjectVersion:
+    Type: String
+  EnvironmentTypeAbbreviation:
+    Type: String
+Resources:
+  # IAM Roles
+  AnalyticsLambdaExecutionIAMRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+            - Effect: "Allow"
+              Principal:
+                Service:
+                  - "lambda.amazonaws.com"
+              Action:
+                - "sts:AssumeRole"
+      Path: "/"
+      ManagedPolicyArns:
+        - "arn:aws:iam::aws:policy/service-role/AWSLambdaKinesisExecutionRole"
+        - "arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole"
+  # Lambda Functions
+  AnalyticsBigqueryLambdaFunction:
+    Type: "AWS::Lambda::Function"
+    Properties:
+      Code:
+        S3Bucket: !Ref CodeS3Bucket
+        S3Key: !Sub "lambda/PRX-analytics-ingest-lambda.zip"
+        S3ObjectVersion: !Ref CodeS3ObjectVersion
+      Description: Dovetail analytics to BigQuery
+      Environment:
+        Variables:
+          BQ_IMPRESSIONS_TABLE: "impressions"
+          BQ_DOWNLOADS_TABLE: "downloads"
+          BQ_PROJECT_ID: "prx-metrics"
```

Comment:
Do we vary any of this by env? - Is this the same project for prod and staging?

Reply:
We use the same "project" for all 3 of our datasets: development, staging and production.

It's actually a bit of a TBD if this will work - having _some_ of the ENVs in cloudformation, but excluding others.  My hope is that CFN will just leave those ENVs alone that it doesn't know about.  We'll find out after this deploys.

The long-term hope is that Lambda gives 1st-class-citizen status to parameter-store.  Then we can just specify a paramstore name in here, and vary it based on staging-vs-prod.